torch/quantization/observer.py

```diff
@@ -1003,6 +1005,28 @@ def calculate_qparams(self):
     def get_qparams(self):
         return self.calculate_qparams()
 
+class TensorListObserver(_ObserverBase):
```

Comment:
@jerryzh168 if I implement the observer for tensor lists this way, it cannot be scripted because of the use of `cls` inside it
```
RuntimeError:
Module 'TensorListObserver' has no attribute '_cls' (This attribute exists on the Python module, but we failed to convert Python type: 'ABCMeta' to a TorchScript type.):
  File "/home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/quantization/observer.py", line 1020
        # type (List[Tensor]) -> List[Tensor]
        for t in input:
            new_obs = self._cls(dtype=dtype, qscheme=qscheme, reduce_range=reduce_range)
                      ~~~~~~~~~ <--- HERE
            new_obs.forward(t)
            self._observers.append(new_obs)
```

Reply:
can you split the computation in observer into a pure function and we can get that function from the `cls` and call it in forward?

Reply:
Discussed offline with Raghu, we decided that it is better to have an internal class for MinMaxObserver that supports list of tensors. Since we will refactor LSTM python API later, we might need to re-consider the type of observer we use then.

Reply:
do you need to change all observers to support list?